administration/src/components/cards/CreateCardsController.tsx

```diff
@@ -35,14 +37,22 @@ const CreateCardsController = () => {
   const confirm = async () => {
     try {
       setState(CardActivationState.loading)
+
       const activationCodes = cardBlueprints.map(cardBlueprint => {
         return cardBlueprint.generateActivationCode()
       })
+      const staticCodes = projectConfig.staticQrCodesEnabled
+        ? cardBlueprints.map(cardBlueprints => {
+            return cardBlueprints.generateStaticVerifyCode()
+          })
+        : null
+
+      const font = await loadTTFFont('NotoSans', 'normal', '/pdf-fonts/NotoSans-Regular.ttf')
+      const pdfDataUri = generatePdf(font, region, activationCodes, staticCodes)
 
       await activateCards(client, activationCodes, region)
 
-      const font = await loadTTFFont('NotoSans', 'normal', '/pdf-fonts/NotoSans-Regular.ttf')
-      const pdfDataUri = generatePdf(font, activationCodes, region)
+      if (staticCodes) await activateCards(client, staticCodes, region)
```

Comment:
I htink you already fixed there, that if the PDF generation fails, then the cards dop not get activated.

Reply:
yes